### PR TITLE
Post window-combination-limit and set-window-combination-limit

### LINKED ISSUE #396: Port window-combination-limit to Rust
```c
DEFUN ("window-combination-limit", Fwindow_combination_limit, Swindow_combination_limit, 1, 1, 0,
       doc: /* Return combination limit of window WINDOW.
WINDOW must be a valid window used in horizontal or vertical combination.
If the return value is nil, child windows of WINDOW can be recombined with
WINDOW's siblings.  A return value of t means that child windows of
WINDOW are never (re-)combined with WINDOW's siblings.  */)
  (Lisp_Object window)
{
  struct window *w;

  CHECK_VALID_WINDOW (window);
  w = XWINDOW (window);
  if (WINDOW_LEAF_P (w))
    error ("Combination limit is meaningful for internal windows only");
  return w->combination_limit;
}
```

### PR DESCRIPTION
**Pull request by shaleh**
Added LispWindowRef::is_internal as well.

Closes #396

### CODE REVIEW

File: rust_src/src/windows.rs [hunk 1]
```diff
@@ -165,9 +170,45 @@ pub fn window_margins(window: LispObject) -> LispObject {
...
+
+    if !w.is_internal() {
+        error!("Combination limit is meaningful for internal windows only");
+    }
+
+    w.combination_limit = limit.to_raw();
+
+    limit
+}
+
```
Comment (shanavas786) [inline rust_src/src/windows.rs]: docstring for minibuffer_selected_window is misplaced

Comment (shaleh) [reply rust_src/src/windows.rs]: Looks like a merge mixup. Good catch.

File: rust_src/src/windows.rs [hunk 2]
```diff
@@ -165,6 +170,41 @@ pub fn window_margins(window: LispObject) -> LispObject {
...
+    LispObject::from_raw(w.combination_limit)
+}
+
+/// Set combination limit of window WINDOW to LIMIT; return LIMIT.
+/// WINDOW must be a valid window used in horizontal or vertical combination.
+/// If LIMIT is nil, child windows of WINDOW can be recombined with WINDOW's
+/// siblings.  LIMIT t means that child windows of WINDOW are never
+/// \(re-)combined with WINDOW's siblings.  Other values are reserved for
+/// future use.
```
Comment (brotzeit) [inline rust_src/src/windows.rs]: You can remove the backslash.

Comment (shaleh) [reply rust_src/src/windows.rs]: Will do

**Comment by shaleh** [comment]
Waiting on PR #411

**Pull request merged**